classes/Hook.php

```diff
@@ -235,11 +236,111 @@ public static function getHookAliasList()
     }
 
     /**
+     * Get the list of hook aliases
+     *
+     * @since 1.7.1.0
+     * @return array
+     */
+    private static function getHookAliasesList()
+    {
+        $cacheId = 'hook_aliases';
+        if (!Cache::isStored($cacheId)) {
+            $hookAliasList = Db::getInstance()->executeS('SELECT * FROM `'._DB_PREFIX_.'hook_alias`');
+            $hookAliases = array();
+            if ($hookAliasList) {
+                foreach ($hookAliasList as $ha) {
+                    if (!isset($hookAliases[$ha['name']])) {
+                        $hookAliases[$ha['name']] = array();
+                    }
+                    $hookAliases[$ha['name']][] = $ha['alias'];
+                }
+            }
+            Cache::store($cacheId, $hookAliases);
+            return $hookAliases;
+        }
+        return Cache::retrieve($cacheId);
+    }
+
+    /**
+     * Return backward compatibility hook names
+     *
+     * @since 1.7.1.0
+     * @param $hookName
+     * @return array
+     */
+    private static function getHookAliasesFor($hookName)
+    {
+        $cacheId = 'hook_aliases_' . $hookName;
+        if (!Cache::isStored($cacheId)) {
+            $aliasesList = Hook::getHookAliasesList();
```

Comment:
You are going to cache something already coming from the cache. We may remove it.